### ISSUE TITLE
Not all violations are displayed on the stackrox dashboard

### ISSUE THREAD

**Issue by VadimPlekhanov** [stackrox/stackrox#8253]
Enabled the Linux User Add Execution policy. We violate the policy by adding users. After adding 5-8 users, violations are not displayed in the dashboard. Tell me what is the reason for the unstable operation? On the nodes where eBPF runs, ubuntu-22 is installed.

**Comment by dvail** [comment]
Hi @VadimPlekhanov - are the users being added to the cluster nodes or to containers running within the cluster?

If the users are being added to the cluster nodes, that will _not_ trigger policy violations. Currently violations are only triggered for this type of policy when the action occurs on a container within the cluster.

If these users are being added to a container within the cluster, can you provide the version of StackRox you are running so that we can attempt to reproduce on our end?

**Comment by VadimPlekhanov** [comment]
Hi @dvail, users are being added to a container within the cluster. We are using stackrox version v4.2.0.

**Comment by dvail** [comment]
Thanks for the update, we are tracking this internally and I'll notify the correct team. CC: @stackrox/core-workflows

**Comment by clickboo** [comment]
@VadimPlekhanov Would you please share the exact command you used to add users to a container? Thanks!

**Comment by VadimPlekhanov** [comment]
@clickboo, "useradd USER1", "useradd USER2" etc...

**Comment by clickboo** [comment]
Thanks @VadimPlekhanov, also just to confirm these commands are not in the dockerfile but being executed inside of a running container of a deployment?

**Comment by VadimPlekhanov** [comment]
yes, of course, these commands are executed inside of a running container.

**Comment by clickboo** [comment]
Thanks, would you be able to share if there are any exclusions configured on the policy and/or scope? These configurations control whether the policy is evaluated on the deployment in question or not, and I would like to confirm that a violation not being generated, is a consequence of the policy configuration of exclusion and/or scope. Pasting the policy yaml here (obviously redacting any sensitive info) would also work.

**Comment by danekantner** [comment]
are those deployments that aren't showing still considered active?

**Comment by VadimPlekhanov** [comment]
@clickboo, this policy has scope for NS. There are no exceptions. I would like to note that the policy works, but not for every event "useradd|usermod".
[StackRox_Exported_Policies-11_17_2023.json](https://github.com/stackrox/stackrox/files/13390079/StackRox_Exported_Policies-11_17_2023.json)

**Comment by VadimPlekhanov** [comment]
@danekantner, yes, these deployments are active

**Comment by fredrb** [comment]
@VadimPlekhanov thanks for reporting the issue and sorry for the delay. 

The issue you're seeing is by design to prevent overloading the system with too many similar process instances. We limit argument prefixes to reduce duplicates - meaning if we get repetitive processes with slight argument changes, they're less likely to be reported over time.

If you're interested in taking a look for yourself, the portion of the code that is responsible for filtring processes has a [fan out property](https://github.com/stackrox/stackrox/blob/master/pkg/process/filter/filter.go#L58), which defines the bucket sizes for each argument position. Bucket sizes for [sensor](https://github.com/stackrox/stackrox/blob/master/sensor/common/processfilter/filter.go#L17) are `8, 6, 4, 2`. Args to the left (binary included) will be allowed higher repetition before filtering. For example, in your case, `useradd` will be reported 8 times for the same container ID before its filtered out. `useradd -D` would take only 6 instances to start being filtered out.

We currently don't have concrete plans for changing this, but it's part of other limitations we have in sensor regarding optimizations of event handling.

**Comment by porridge** [comment]
@VadimPlekhanov does @fredrb 's description answer your issue? Can this be closed? Do you think a documentation update would be helpful?

**Comment by msugakov** [comment]
A potential improvement to not filter (or reduce filtering) has been backlogged. I can't see any committed date on that. I suppose we keep this issue open to share any updates on progress.
